### ISSUE TITLE
AppIcon not displayed for Android 7

### ISSUE THREAD

**Issue by Syed-RI** [dotnet/maui#23178]
### Description

App icon in the launcher and task switcher is not displaying for Android 7. However, Android 8 and up works
![Task_Switcher_Android8](https://github.com/dotnet/maui/assets/100694732/7fff8ed3-edc8-4020-b572-b56ef850bd9c)
![AppIcon_Android8](https://github.com/dotnet/maui/assets/100694732/9ac6ac5d-7423-4606-bac5-ec5f7d2cede7)
![Task_Switcher_Android7](https://github.com/dotnet/maui/assets/100694732/adb58724-9f6d-492e-bf1b-38b6257d459f)
![AppIcon_Android7 png](https://github.com/dotnet/maui/assets/100694732/fbf61584-c879-4222-bef1-6e3a2af6a55c)

### Steps to Reproduce

Create a new maui app and deploy to an emulator running Android 7 (7.1.1) with a screen resolution of 1280 x 720. Or build and deploy the repro attached to same emulator.

### Link to public reproduction project repository

https://github.com/Syed-RI/DeployTest

### Version with bug

8.0.60 SR6

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 7.1.1

### Did you find any workaround?

Works on Android 8 and up

**Comment by Equabyte** [comment]
@Syed-RI I have an app in production compiled with 8.0.60 SR6 that is tested on physical devices with all supported Android versions including Android 7 and the app icon is showing correctly (despite its look not being the same as in latest Android versions). Image below is from a **Samsung A3 2016** running Android 7 (last Android version supported on that device). So this issue seems either device-specific or emulator related.
![DCF_Calc_Android_7](https://github.com/dotnet/maui/assets/145383043/f7423ab4-8a85-4b09-a310-d000aeff6587)

**Comment by Syed-RI** [comment]
It happens on physical device as well. I think it is probably related to the screen resolution mine was 1280 x 720.

**Comment by Equabyte** [comment]
> It happens on physical device as well. I think it is probably related to the screen resolution mine was 1280 x 720.

Galaxy A3 (2016) is also 720 x 1280 so it does not seem to be the resolution.

**Comment by Equabyte** [comment]
And here's your icon on my Galaxy A3 (2016) running Android 7: 

![deploy_test_icon](https://github.com/dotnet/maui/assets/145383043/0baffda6-dd7b-4e28-b6bc-5ecff0056869)

**Comment by Syed-RI** [comment]
> And here's your icon on my Galaxy A3 (2016) running Android 7: 
> 
> ![deploy_test_icon](https://github.com/dotnet/maui/assets/145383043/0baffda6-dd7b-4e28-b6bc-5ecff0056869)
> 

Plot thickens

**Comment by Equabyte** [comment]
> Plot thickens

There is a caveat though: Samsung Android 7 devices use an icon container by default (visible in my image) which may hide the true issue you are facing on your device/emulator. So indeed worth checking further in case Android 7 is really important to you; but according to [this chart](https://gs.statcounter.com/os-version-market-share/android) the share of Nougat users is now less than 1.5% worldwide.

**Comment by Zhanglirong-Winnie** [comment]
This issue has been verified using Visual Studio 17.11.0 Preview 2.1 (8.0.60 & 8.0.40 & 8.0.10).  Can be reproduced on Android(7.1 &1280 x 720)platform.
![image](https://github.com/dotnet/maui/assets/77434865/7afa020e-067b-42ea-b386-221d36c7e4e9)

**Comment by Syed-RI** [comment]
>There is a caveat though: Samsung Android 7 devices use an icon container by default (visible in my image) which may hide the true issue you are facing on your device/emulator. So indeed worth checking further in case Android 7 is really important to you; but according to [this chart](https://gs.statcounter.com/os-version-market-share/android) the share of Nougat users is now less than 1.5% worldwide.

This is for an enterprise app and we are committed to provide support for Android 7 and up. Besides, MAUI SDK itself supports Android 7 and up therefore I think this should be looked at.

**Comment by mattleibow** [comment]
I'll have a look. This is very weird as we have not really changed anything and there should be a drawable fallback.

**Comment by mattleibow** [comment]
OK, I managed to repro it... Not sure the issue yet as the images are there and the image works for Android 6.0.

I typically test Android 6, 9, 10, 13, 14 since they are on borders of major OS changes. They all work on those I believe, so Android 7 is giving us gifts of joy it seems.

**Comment by mattleibow** [comment]
I had a look and I see why this is happening:

 - Android 6 only supported "square" icons
 - Android 7 added round icons - https://developer.android.com/about/versions/nougat/android-7.1#circular-icons
 - Android 8 switched to adaptive icons - https://developer.android.com/studio/write/create-app-icons#launcher

This means that specifically for Android 7, the round icons are PNG files but for Android 8 they are adaptive icons.

So, the first workaround is to just drop round icons altogether. This means you will lose round icons on **all** Android versions, but just involves deleting the `android:roundIcon="@mipmap/appicon_round"` from your `Platforms/Android/AndroidManifest.xml`

If you still want round icons for new Android versions and maybe less concerned with the Android 7 app icon adherence to the "round" design, you could copy the generated icons from `obj/Debug/net8.0-android/resizetizer/r/mipmap-*/appicon.png` and add them to `Platforms/Android/Resources/mipmap-*/appicon_round.png`. Then in your csproj, you can add them as Android-specific resources: `<AndroidResource Include="Platforms\Android\Resources\mipmap-*\*" />`

This will just be a copy of the "legacy" icons and pretend they are round. But, if that is not viable, then you can potentially generate the round ones using a graphics program and place them in the same location. As long as you name it all the same as the other icons, they will just work together.

As soon as I have a fix for generating legacy round icons, I will link to it.

**Comment by Syed-RI** [comment]
> I had a look and I see why this is happening:
> 
>  - Android 6 only supported "square" icons
>  - Android 7 added round icons - https://developer.android.com/about/versions/nougat/android-7.1#circular-icons
>  - Android 8 switched to adaptive icons - https://developer.android.com/studio/write/create-app-icons#launcher
> 
> This means that specifically for Android 7, the round icons are PNG files but for Android 8 they are adaptive icons.
> 
> So, the first workaround is to just drop round icons altogether. This means you will lose round icons on **all** Android versions, but just involves deleting the `android:roundIcon="@mipmap/appicon_round"` from your `Platforms/Android/AndroidManifest.xml`
> 
> If you still want round icons for new Android versions and maybe less concerned with the Android 7 app icon adherence to the "round" design, you could copy the generated icons from `obj/Debug/net8.0-android/resizetizer/r/mipmap-*/appicon.png` and add them to `Platforms/Android/Resources/mipmap-*/appicon_round.png`. Then in your csproj, you can add them as Android-specific resources: `<AndroidResource Include="Platforms\Android\Resources\mipmap-*\*" />`
> 
> This will just be a copy of the "legacy" icons and pretend they are round. But, if that is not viable, then you can potentially generate the round ones using a graphics program and place them in the same location. As long as you name it all the same as the other icons, they will just work together.
> 
> As soon as I have a fix for generating legacy round icons, I will link to it.

Awesome Matt!

We can live without the round icons so I'll look into it tomorrow. But IIRC, I have tried copying and pasting from resizetizer output and that broke the build as on next build resizetizer complained about same assets (i.e. appicon.png) being present in the dir.

**Comment by mattleibow** [comment]
> resizetizer complained about same assets 

Yeah, make sure to copy the `appicon` and rename to `appicon_round`. You will just need those variants as all the others are still generated.

**Comment by mattleibow** [comment]
This PR has the fixes to generate round icons for Android 7 https://github.com/dotnet/maui/pull/23276
